liquibase-cli/src/main/java/liquibase/integration/commandline/CommandRunner.java

```diff
@@ -0,0 +1,75 @@
+package liquibase.integration.commandline;
+
+import liquibase.Scope;
+import liquibase.command.CommandResults;
+import liquibase.command.CommandScope;
+import liquibase.integration.IntegrationConfiguration;
+import liquibase.util.StringUtil;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+class CommandRunner implements Callable<CommandResults> {
+
+    private CommandLine.Model.CommandSpec spec;
+
+    @Override
+    public CommandResults call() throws Exception {
+        List<String> command = new ArrayList<>();
+        command.add(spec.commandLine().getCommandName());
+
+        CommandLine parentCommand = spec.commandLine().getParent();
+        while (!parentCommand.getCommandName().equals("liquibase")) {
+            command.add(0, parentCommand.getCommandName());
+            parentCommand = parentCommand.getParent();
+        }
+        final CommandLine rootCommand = parentCommand;
+
+
+        final String[] commandName = LiquibaseCommandLine.getCommandNames(spec.commandLine().getParseResult());
+        for (int i=0; i<commandName.length; i++) {
+            commandName[i] = StringUtil.toCamelCase(commandName[i]);
+        }
+
+        final CommandScope commandScope = new CommandScope(commandName);
+        final File outputFile = IntegrationConfiguration.OUTPUT_FILE.getCurrentValue();
+        OutputStream outputStream = null;
+
+        try {
```

Comment:
Should we be using try with resources here?

Reply:
The issue with doing a try-with here is that the OutputStream need to exist for the scope of the command, so it will need to be close after the command finishes.

Reply:
I think it's the "if" statement part that doesn't work with try with resources.